OpenSim/Sandbox/ExampleHopperDevice/helperMethods.h

```diff
@@ -0,0 +1,254 @@
+
+
+//------------------------------------------------------------------------------
+// Configure a PathActuator so that it wraps over a WrapObject attached to the
+// specified Body. The footwork in this method is necessary because of a bug in
+// GeometryPath.
+//------------------------------------------------------------------------------
+inline void addPathWrapHelper(ModelComponent& model,
+    const std::string& pathActuatorName, const std::string& wrapObjectName,
+    const std::string& bodyName);
+
+
+//------------------------------------------------------------------------------
+// Build a testbed for testing the device before attaching it to the hopper. We
+// will attach one end of the device to ground ("/testbed/ground") and the other
+// end to a sprung load ("/testbed/load").
+//------------------------------------------------------------------------------
+inline Model buildTestbed();
+
+
+//------------------------------------------------------------------------------
+// SignalGenerator is a type of Component with no inputs and only one output.
+// This Component evaluates an OpenSim::Function (stored in its "function"
+// property) as a function of time. We can use a SignalGenerator to design
+// time-varying control inputs for testing the device.
+//------------------------------------------------------------------------------
+class SignalGenerator : public Component {
+    OpenSim_DECLARE_CONCRETE_OBJECT(SignalGenerator, Component);
+
+public:
+    OpenSim_DECLARE_PROPERTY(function, Function,
+        "Function used to generate the signal (a function of time)");
+    OpenSim_DECLARE_OUTPUT(signal, double, getSignal, SimTK::Stage::Time);
+
+    SignalGenerator() { constructProperties(); }
+
+    double getSignal(const SimTK::State& s) const {
+        return get_function().calcValue(SimTK::Vector(1, s.getTime())); }
+
+private:
+    void constructProperties() { constructProperty_function(Constant(0.)); }
+
+}; // end of SignalGenerator
+
+
+//==============================================================================
+//                               IMPLEMENTATIONS
+//==============================================================================
+template <class C>
+inline void showSubcomponentInfo(const Component& comp)
+{
+    using std::cout; using std::endl;
+
+    std::string className = typeid(C).name();
+    const std::size_t colonPos = className.rfind(":");
+    if (colonPos != std::string::npos)
+        className = className.substr(colonPos+1, className.length()-colonPos);
+
+    cout << "Class name and full path name for descendants of '"
+         << comp.getName() << "' of type " << className << ":\n" << endl;
+
+    ComponentList<C> compList = comp.getComponentList<C>();
+
+    // Step through compList once to find the longest concrete class name.
+    unsigned maxlen = 0;
+    for (const C& thisComp : compList)
+        maxlen = std::max(maxlen, thisComp.getConcreteClassName().length());
```

Comment:
I am getting an error on this line due to mismatch of types of arguments to `std::max`. First one is *deduced* to be `unsigned int`, second one is *deduced* to be `unsigned long`.